packages/contracts-bedrock/test/kontrol/proofs/OptimismPortal.k.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.8.13;
+
+import { DeploymentSummary } from "./utils/DeploymentSummary.sol";
+import { KontrolUtils } from "./utils/KontrolUtils.sol";
+import { Types } from "src/libraries/Types.sol";
+import {
+    IOptimismPortal as OptimismPortal,
+    ISuperchainConfig as SuperchainConfig
+} from "./interfaces/KontrolInterfaces.sol";
+
+contract OptimismPortalKontrol is DeploymentSummary, KontrolUtils {
+    OptimismPortal optimismPortal;
+    SuperchainConfig superchainConfig;
+
+    function setUp() public {
+        recreateDeployment();
+        optimismPortal = OptimismPortal(payable(OptimismPortalProxyAddress));
+        superchainConfig = SuperchainConfig(SuperchainConfigProxyAddress);
+    }
+
+    /// TODO: Replace struct parameters and workarounds with the appropiate
+    /// types once Kontrol supports symbolic `bytes` and `bytes[]`
+    function test_proveWithdrawalTransaction_paused(
```

Comment:
One thought I had: It seems this test uses symbolic calldata, but concrete storage? The code paths in the `proveWithdrawalTransaction` and `finalizeWithdrawalTransaction` methods should not branch based on any storage except for the `whenNotPaused` modifier. Therefore, what are your thoughts on modeling storage as symbolic here too? Since there's no additional branches I'd think the performance hit is minimal, and it gives assurance that we didn't accidentally add any branching logic that accidentally bypasses the pause. I can also see how this might be overkill though, so need to think on it more, but wanted to get your thoughts

Reply:
Do you mean having the storage of the `OptimismPortal` be symbolic? We can try that! The first difficulty that comes to mind is that, if I'm correct, the cheatcode for making the storage symbolic will override any previously written value to the storage of the account. Hence overriding the initialization (which is necessary not to abstract away). But that would only be if we invoke the cheatcode in the test. We might try setting the storage of `OptimismPortal` symbolic before calling the summary contract's `recreateDeployment` function.
However, how well would it play with the automatic generation of the KEVM initial state from the deployment script? It's unclear I think. Because if we run the summary contract with Kontrol, we can choose when to run it. But with the KEVM init state generation, we "run" `recreateDeployment` before Kontrol even executes any test file.
I think this is a nice item to investigate!

Reply:
> Do you mean having the storage of the `OptimismPortal` be symbolic? 

Yes exactly! To your points, I guess this makes me wonder if we were going down the wrong path on the deploy script / initial state setup? In other words, how much should we actually care about the result of the deploy script, if making all storage symbolic (with paused state as the exception) is more generic and simplifies setup? (Assuming not a huge performance hit). 

We don't have to necessarily answer this on this PR, but agreed this is worth thinking about and investigating. So I'll resolve this comment and move the discussion out of this PR